spec/helpers/friends_helper_spec.rb

```diff
@@ -15,7 +13,7 @@
 
     it 'returns the last checkin address if it exists' do
       expect(helper.friends_device_last_checkin(device.checkins.first)).to match('Last available')
-      expect(helper.friends_device_last_checkin(device.checkins.first)).to match(device.checkins.first.fogged_city.to_s)
+      expect(helper.friends_device_last_checkin(device.checkins.first)).to match(device.checkins.first.city.to_s)
```

Comment:
The checkin that is passed to the helper is not a regular checkin object though. As this is the friends page, the checkin will have been returned from safe_checkin_info_for, in fact this method should probably be rewritten as follows:
```
def friends_device_last_checkin(device)
  checkin = device.safe_checkin_info_for(permissible: current_user, action: 'last')[0]
  if checkin.present?
    "<p>Last available reported area: #{checkin['city']}</p>".html_safe
  else
    '<p>No location found</p>'.html_safe
  end
end
```

In safe_checkin_info_for it is possible for the checkin.city to be replaced with fogged_city.